models/Ion_auth_model.php

```diff
@@ -2499,4 +2338,36 @@ protected function _filter_data($table, $data)
 	}
 
 	/**
+
+	/** Generate a random token
+	 * Inspired from http://php.net/manual/en/function.random-bytes.php#118932
+	 *
+	 * @param int $result_length
+	 * @return string
+	 */
+	protected function _random_token($result_length = 32)
+	{
+		if(!isset($result_length) || intval($result_length) <= 8 ){
+			$result_length = 32;
+		}
+
+		// Try random_bytes: PHP 7
+		if (function_exists('random_bytes')) {
+			return bin2hex(random_bytes($result_length / 2));
+		}
+
+		// Try mcrypt
+		if (function_exists('mcrypt_create_iv')) {
+			return bin2hex(mcrypt_create_iv($result_length / 2, MCRYPT_DEV_URANDOM));
+		}
+
+		// Try openssl
+		if (function_exists('openssl_random_pseudo_bytes')) {
+			return bin2hex(openssl_random_pseudo_bytes($result_length / 2));
+		}
+
+		// No luck!
+		show_error("Cannot generate a token random enough. " .
```

Comment:
What you have here is good, but add these function checks and the error to the library compat check as well.  That way the user sees up front that they arent compatibile instead of it popping up when they test the token functionality.

Reply:
Got it.